storage/remote/azuread/azuread.go

```diff
@@ -146,21 +193,46 @@ func (rt *azureADRoundTripper) RoundTrip(req *http.Request) (*http.Response, err
 
 // newTokenCredential returns a TokenCredential of different kinds like Azure Managed Identity and Azure AD application.
 func newTokenCredential(cfg *AzureADConfig) (azcore.TokenCredential, error) {
-	cred, err := newManagedIdentityTokenCredential(cfg.ManagedIdentity.ClientID)
+	var cred azcore.TokenCredential
+	var err error
+	cloudConfiguration, err := getCloudConfiguration(cfg.Cloud)
 	if err != nil {
 		return nil, err
 	}
+	clientOpts := &azcore.ClientOptions{
+		Cloud: cloudConfiguration,
+	}
+
+	if cfg.ManagedIdentity != nil {
+		cred, err = newManagedIdentityTokenCredential(clientOpts, cfg.ManagedIdentity.ClientID)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if cfg.OAuth != nil {
+		cred, err = newOAuthTokenCredential(clientOpts, cfg.OAuth.TenantID, cfg.OAuth.ClientID, cfg.OAuth.ClientSecret)
+		if err != nil {
+			return nil, err
+		}
+	}
 
 	return cred, nil
 }
 
 // newManagedIdentityTokenCredential returns new Managed Identity token credential.
-func newManagedIdentityTokenCredential(managedIdentityClientID string) (azcore.TokenCredential, error) {
+func newManagedIdentityTokenCredential(clientOpts *azcore.ClientOptions, managedIdentityClientID string) (azcore.TokenCredential, error) {
 	clientID := azidentity.ClientID(managedIdentityClientID)
-	opts := &azidentity.ManagedIdentityCredentialOptions{ID: clientID}
+	opts := &azidentity.ManagedIdentityCredentialOptions{ClientOptions: *clientOpts, ID: clientID}
 	return azidentity.NewManagedIdentityCredential(opts)
 }
 
+// newOAuthTokenCredential returns new OAuth token credential
+func newOAuthTokenCredential(clientOpts *azcore.ClientOptions, tenantID, clientID, clientSecret string) (azcore.TokenCredential, error) {
```

Comment:
Could this function just take one `*OAuthConfig{}` instead of all the fields separately? And you might as well do the same with `newManagedIdentityTokenCredential` for uniformity.